evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -258,6 +264,7 @@ pub fn apply<DB: DatabaseExt>(
             Bytes::new()
         }
         HEVMCalls::Deal(inner) => {
+            ensure!(!is_potential_precompile(inner.0), "Deal cannot be used on precompile addresses (N < 10). Please use an address bigger than 10 instead");
```

Comment:
does this result in a revert? if so we might want to add some `vm.expectRevert` tests for these

Reply:
added tests in https://github.com/foundry-rs/foundry/pull/4905/commits/26fe2ebf5a64588250e81c709d3ae538175733ae , also removed the revert on deal